src/object/schema_template.c

```diff
@@ -4066,12 +4066,18 @@ int
 smt_drop_query_spec (SM_TEMPLATE * def, const int index)
 {
   int error = NO_ERROR;
-  SM_QUERY_SPEC *file, *prev, *found;
+  SM_QUERY_SPEC *file, *prev, *found, *user_query_spec;
   int i;
   char indexname[20];
 
-  for (file = def->query_spec, prev = NULL, found = NULL, i = 1; file != NULL && found == NULL; file = file->next, i++)
+  for (file = def->query_spec, prev = NULL, found = NULL, i = 1; file != NULL && found == NULL; file = file->next)
```

Comment:
I think it can be simplified as below.

```
for (file = def->query_spec, prev = NULL, found = NULL, i = 1; file != NULL;  i++)
    {
      if (index == i)
	{
	  found = file;
          break;
	}
       assert(file->next != NULL && file->next->specification[0] == '-')
       prev = file->next;
       file = file->next->next;
    }
```